descriptors.go

```diff
@@ -54,29 +55,43 @@ func (d *descriptorSet) add(descs ...*descriptor.FileDescriptorProto) {
 //
 // This is equivalent to the following command:
 //
-// cat merged.pb | protoc --decode google.protobuf.FileDescriptorSet /usr/local/include/google/protobuf/descriptor.proto
-func (d *descriptorSet) marshalTo(w io.Writer) error {
+// cat merged.pb | protoc --decode google.protobuf.FileDescriptorSet /path/to/google/protobuf/descriptor.proto
+func (d *descriptorSet) marshalTo(w io.Writer, paths []string) error {
 	p, err := proto.Marshal(&d.merged)
 	if err != nil {
 		return err
 	}
 
+	const descProto = "google/protobuf/descriptor.proto"
```

Comment:
This search code should be outside of `marshalTo`. Could we just take this path as an argument to `newDescriptorSet`?

Reply:
I have to admit I don't fully get it. Do you mean something like this (please see updated commit)?

If not, maybe you can create something based on the code/idea in this patch?